pennylane/operation.py

```diff
@@ -1875,22 +1877,15 @@ def is_hermitian(self):
         """All observables must be hermitian"""
         return True
 
-    # pylint: disable=abstract-method
-    return_type = None
-    """None or ObservableReturnTypes: Measurement type that this observable is called with."""
-
-    def __repr__(self):
-        """Constructor-call-like representation."""
-        temp = super().__repr__()
-
-        if self.return_type is None:
-            return temp
-
-        return (
-            f"{repr(self.return_type)}(wires={self.wires.tolist()})"
-            if self.return_type is qml.measurements.Probability
-            else f"{repr(self.return_type)}({temp})"
+    @property
+    def return_type(self):
+        """None or ObservableReturnTypes: Measurement type that this observable is called with."""
+        warnings.warn(
+            "`Observable.return_type` is deprecated. Instead, you should "
+            "inspect the type of the surrounding measurement process.",
```

Comment:
Could we add an example?

Reply:
uhh. kinda but I'm not sure what it would look like? need a tape to explain why they might be different IMO
```pycon
>>> tape = qml.tape.QuantumScript([], [qml.expval(qml.PauliZ(0))])
>>> meas, obs = tape.measurements[0], tape.observables[0]
>>> print(meas, obs)
expval(PauliZ(wires=[0])) PauliZ(wires=[0])
>>> print(meas.return_type, obs.return_type)
pennylane/operation.py:1883: UserWarning: `Observable.return_type` is deprecated. Instead, you should inspect the type of the surrounding measurement process.
  warnings.warn(
ObservableReturnTypes.Expectation ObservableReturnTypes.Expectation
```

Reply:
Thanks! Ok, let's leave it for now.